chia/wallet/wallet_state_manager.py

```diff
@@ -1655,9 +1700,10 @@ async def create_wallet_for_puzzle_info(self, puzzle_driver: PuzzleInfo, name=No
                 name,
             )
 
-    async def add_new_wallet(self, wallet: WalletProtocol) -> None:
+    async def add_new_wallet(self, wallet: WalletProtocol, create_puzhash: bool = True) -> None:
         self.wallets[wallet.id()] = wallet
-        await self.create_more_puzzle_hashes()
+        if create_puzhash:
```

Comment:
This is already handled via `WalletProtocol.require_derivation_paths`

Reply:
noted, removed